webrtc_ros/CMakeLists.txt

```diff
@@ -69,42 +72,48 @@ target_link_libraries(${PROJECT_NAME}_server_noncpp11
   ${catkin_LIBRARIES}
 )
 
+
+
 # This library is built with the same options used to build the webrtc library
 add_library(${PROJECT_NAME}_server
   src/configure_message.cpp
   src/ice_candidate_message.cpp
   src/image_transport_factory.cpp
-  src/ros_log_context.cpp
   src/ros_video_capturer.cpp
   src/ros_video_renderer.cpp
   src/sdp_message.cpp
   src/webrtc_client.cpp
   src/webrtc_ros_message.cpp
   src/webrtc_ros_server.cpp
+  src/webrtc_ros_json_parser.cpp
 )
 add_dependencies(${PROJECT_NAME}_server ${catkin_EXPORTED_TARGETS} ${${PROJECT_NAME}_EXPORTED_TARGETS})
 target_link_libraries(${PROJECT_NAME}_server
   ${catkin_LIBRARIES}
   ${webrtc_LIBRARIES}
   ${X11_LIBRARIES}
-  ${PROJECT_NAME}_server_noncpp11
+  jsoncpp_object
```

Comment:
We are you linking against the pseudo OBJECT library? You should probably link against the regular shared library, and this would likely solve your wrong ABI problem.

Reply:
> Thank you for this awesome pull request! At first glance, your changes look good to me. Since the current build in `develop` is broken anyway due to the changes in `depot_tools`, I suggest I'll merge this PR as-is, and then we iron out the remaining problems in follow-up PRs, particularly the `libjsoncpp` issue. For the `async_web_server_cpp`, we'll have to revive [GT-RAIL/async_web_server_cpp](https://github.com/GT-RAIL/async_web_server_cpp); I don't have write access to that repository, so we'll have to find someone who has. Failing that, we can try to fork the repository and hijack the package, but I'd like to avoid that if possible.

That sounds good! Thanks! I think there will be more work to make it work for arm. 

> We are you linking against the pseudo OBJECT library? You should probably link against the regular shared library, and this would likely solve your wrong ABI problem.

You are right! This is not the right way to do it. Just to be clear, this was a rather temporary solution and we should probably figure out why libjsoncpp is not directly built (or doesn't link) with the `webrtc` package. @flynneva made in fact a fix on that in [this pull request](https://github.com/kubja/webrtc_ros/pull/1). It however doesn't work for me with the Ubuntu 20.04 `libjsoncpp-dev` package most likely because of the older version of the library (originally, I thought it is the ABI as I was facing json-related issues with ABI when switching to C++14).